isabelle/Doc/Specification/Guarantees.thy

```diff
@@ -3,70 +3,153 @@ theory Guarantees
   imports       
       Core.SemanticsTypes 
       Core.MoneyPreservation 
+      Core.Timeout
       Core.QuiescentResult
       Core.SingleInputTransactions
-      Core.Timeout
+      Core.CloseSafe
       Core.TransactionBound
  
 begin
 (*>*)
 
 chapter \<open>Marlowe Guarantees\<close>
-text \<open>TODO: add human readable version of the important theorems and lemmas\<close>
+
+text \<open>We can also use proof assistants to demonstrate that the Marlowe semantics presents certain
+desirable properties, such as that money is preserved and anything unspent is returned to users by
+the end of the execution of any contract.\<close>
+
+subsubsection \<open>Auxillary Functions\label{sec:playTrace}\<close>
+
+text \<open>Many of the proofs in this chapter rely on function @{const playTrace} and
+@{const playTraceAux} that execute a sequence of transactions using the Marlowe semantics defined in
+@{const computeTransaction}. They also rely on starting from a valid and positive contract state,
+@{const validAndPositive_state} and a function @{const maxTimeContract} that extracts the latest
+timeout from the contract.\<close>
+
+text \<open>@{const playTrace} :: @{typeof playTrace}\<close>
+text \<open>@{const playTraceAux} :: @{typeof playTraceAux}\<close>
+text \<open>@{const validAndPositive_state} :: @{typeof validAndPositive_state}\<close>
+text \<open>@{const maxTimeContract} :: @{typeof maxTimeContract}\<close>
 
 section \<open>Money Preservation\<close>
-text \<open>TODO: Money preservation\<close>
+
+text \<open>
+
+One of the dangers of using smart contracts is that a badly written one can potentially lock its
+funds forever. By the end of the contract, all the money paid to the contract must be distributed
+back, in some way, to a subset of the participants of the contract. To ensure this is the case we
+proved two properties: "Money Preservation" and "Contracts Always Close".
+
+Regarding money preservation, money is not created or destroyed by the semantics. More specifically,
+the money that comes in plus the money in the contract before the transaction must be equal to the
+money that comes out plus the contract after the transaction, except in the case of an error.
+
+\<close>
+
 text \<open>@{thm playTrace_preserves_money}\<close>
 
+text \<open>where @{const moneyInTransactions} and @{const moneyInPlayTraceResult} measure the funds in
+the transactions applied to a contract versus the funds in the contract state and the payments that
+it has made while executing.\<close>
+
+section \<open>Contracts Always Close\<close>
+
+text \<open>
+
+For every Marlowe Contract there is a time after which an empty transaction can be issued that will
+close the contract and refund all the money in its accounts.
+
+FIXME: This theorem doesn't actually prove the narrative. Are we missing a theorem?
```

Comment:
You are right, it doesn't prove that money is refunded, it proves that the accounts are empty after the transaction. It follows from the money preservation theorem that money was payed out and not burned, but it is tricky to express the "money is refunded" as a theorem. I will try to write another lemma